api/users/views.py

```diff
@@ -529,11 +529,14 @@ def get_default_odm_query(self):
 
     def get_queryset(self):
         nodes = Node.find(self.get_query_from_request())
+        preprints = []
         # TODO: Rearchitect how `.is_preprint` is determined,
         # so that a query that is guaranteed to return only
-        # preprints can be contructed. Use generator in meantime.
-        return (node for node in nodes if node.is_preprint)
-
+        # preprints can be contructed.
+        for node in nodes:
+            for preprint in node.preprints:
```

Comment:
I know there's the todo, and it's not your doing, but this makes me very unhappy. Don't change it obvs, but we do need a Critical-priority ticket for it and label the TODO with the ticket info. This will kill us in the near future unless it all just goes away with DjangOSF.